plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -589,6 +589,27 @@ private void testSelectOrPartitionedByUuid(boolean partitioned)
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testNestedUuid()
+    {
+        String tableName = "test_nested_uuid";
+        assertUpdate(format("CREATE TABLE %s (bigint_t bigint, row_t row(uuid_t uuid, bigint_t bigint), map_t map(bigint, uuid), array_t array(uuid))", tableName));
+
+        String uuid = "UUID '406caec7-68b9-4778-81b2-a12ece70c8b1'";
+        String value = format("VALUES (2, row(%s, 1), map(array[1], array[%1$s]), array[%1$s, %1$s])", uuid);
+        assertUpdate(format("INSERT INTO %s %s", tableName, value), 1);
+
+        assertQuery("SELECT row_t.bigint_t, row_t.uuid_t FROM " + tableName,
+                "VALUES (1, CAST('406caec7-68b9-4778-81b2-a12ece70c8b1' AS UUID))");
+        assertQuery("SELECT map_t[1] FROM " + tableName,
+                "VALUES CAST('406caec7-68b9-4778-81b2-a12ece70c8b1' AS UUID)");
+        assertQuery("SELECT array_t FROM " + tableName,
+                "VALUES ARRAY[CAST('406caec7-68b9-4778-81b2-a12ece70c8b1' AS UUID), CAST('406caec7-68b9-4778-81b2-a12ece70c8b1' AS UUID)]");
```

Comment:
use 

```
assertThat(query(....))
  .matches("VALUES ...");
```

-- this runs control query on Trino, instead of h2

then, use `UUID '...'` literal form instead of CASTs.